pkg/composer/up_service.go

```diff
@@ -165,10 +165,15 @@ func (c *Composer) upServiceContainer(ctx context.Context, service *serviceparse
 		cmd.Stdin = os.Stdin
 	}
 
+	// Containers are always detached, even for `nerdctl compose up --detach=false`.
```

Comment:
This message is confusing. containers created with `nerdctl  run` may be detached or not but compose should always read from container std streams no matter the underlying container is detached or not . In other worder `detach` option should have no effect on the compose level but should keep the standard effect on underlying `nerdcrl run` cmd

Reply:
- `nerdctl compose up --detach=true`: launches `nerdctl run -d`
- `nerdctl compose up --detach=false`: launches `nerdctl run -d` (same as above), and then "attach" `nerdctl logs`

Reply:
Detach should be true to run nerdctl with `-d`  (see https://github.com/containerd/nerdctl/blob/4d9133f76f36efac6115aee2806c67a9b2ef2e8c/pkg/composer/up_service.go#L144)

Reply:
`nerdctl compose up --detach=false` : why `nerdctl run -d` then attach the log viewer . Why not `nerdctl run` and connect cmd std streams to standard std streams ?

Reply:
To allow `nerdctl compose logs` after running `nerdctl compose up --detach=false`

Reply:
OK I forgot that we enforced to write to log file even without `-d` . But I am not agreed with the fact that the underlying `nerdctl run` cmd is always running with the `-d` flag.

Reply:
You're right, the `container.Detached` is always true, but `nerdctl run` isn't always launched with `-d`.

I guess we should just remove the `container.Detached` struct field.

Reply:
Updated

Reply:
that means `container.Detached` have no effect. Users can no longer drop stdout by setting `container.Detached` into the yaml definition ( I believe  it is not compatible with docker too)

I think we should keep `container.Detached` but not always true and then forward  cmd std streams to os std.

that means:

- `container.detached = true` :  `nerdctl run -d ` will log nothing => os std streams will receive nothing
- `container.detached = false` :  `nerdctl run` may log data => os std streams will receive data

```
	if container.Detached && !service.Unparsed.StdinOpen && !service.Unparsed.Tty {
		container.RunArgs = append([]string{"-d"}, container.RunArgs...)
	}

	if service.Unparsed.StdinOpen != service.Unparsed.Tty {
		return "", fmt.Errorf("currently StdinOpen(-i) and Tty(-t) should be same")
	}
	if service.Unparsed.StdinOpen {
		cmd.Stdin = os.Stdin
	}
	cmd.Stdout = os.Stdout
	cmd.Stderr = os.Stderr
```

Reply:
`Detached` was never settable via the YAML AFAICS
https://github.com/compose-spec/compose-spec/blob/master/spec.md

Reply:
ok but strange! LGTM in this case